### PR TITLE
Keywords highlighter

### PR DESCRIPTION
**Pull request by heron2014**
Related: [issue 68](https://github.com/FAC-GM/app/issues/68)

Highlight keywords on the home page: 

keywords

Highlight keywords on the candidate profile page:

candidateprofile

@nelsonic Do you mind to check it?

### CODE REVIEW

File: views/candidateView.html [hunk 1]
```diff
@@ -20,12 +20,9 @@
       {{/if}}
 
       <img src="{{picture}}" alt="" />
-      <h1 class="fullname"><strong>{{fullname}}</strong></h1>
-      <p>{{location}}</p>
-      {{#each tags}}
-        {{{tagsHelper this}}}
-      {{/each}}
-
+      <h1 class="fullname"><strong class="check-highlight">{{fullname}}</strong></h1>
```
Comment (nelsonic) [inline views/candidateView.html]: There is a DRY way of doing this.
however, this is explicit and fine for testing the feature.

File: views/candidateView.html [hunk 2]
```diff
@@ -84,7 +81,7 @@ <h1 class="fullname"><strong>{{fullname}}</strong></h1>
           {{/each}}
         </ul>
     </div>
-
+    <input type="hidden" class="keywords" value="{{keywords}}">
```
Comment (nelsonic) [inline views/candidateView.html]: Any reason you're not getting the keywords from the `search` input?

**Comment by nelsonic** [comment]
I would prefer not to have to add the `class="check-highlight"` to all the templates.
Would prefer the `document.getElementsByClassName('check-highlight');` [**L33**](https://github.com/FAC-GM/app/blob/072871d81b01cb38f9144fe7d736e85031c2a2bb/assets/js/main.js#L33) search through the *existing* classes e.g: `mobile` but as we discussed, we would need to exclude the HTML tag words from the keywords to avoid "breaking" the code... this is something we could add to the https://github.com/dwyl/search-result-keyword-highlighter (*with tests*) to avoid complicating the **App** code...

File: views/candidateView.html (hunk 2 above)
Comment (SimonLab) [inline views/candidateView.html]: Yes I think we could do this and avoid creating a new hidden input.

**Comment by iteles** [comment]
I agree with @nelsonic here. Adding spans with the `check-highlight` class severely pollutes the code. We need to keep it as clean as possible and deliver something we're all happy with (rather than something that feels very hacky and non-scalable).
_Having said this:_ The highlighting is **the lowest priority item on our list** so I would advise that we park this issue for now and work on the higher priority items (particularly whilst you have @nelsonic's availability)

**Comment by nelsonic** [comment]
@iteles in light of the *relative* priority and the fact that this feature is *working*, I think we can merge it and re-factor if we chose to *expand* or *enhance* it later. :+1:

**Pull request merged**